### ISSUE TITLE
loss increasing

### ISSUE THREAD

**Issue by dox012** [facebookresearch/coconut#16]
![Image](https://github.com/user-attachments/assets/d9c2eb05-5311-4d9b-bf53-84798e25547e)
![Image](https://github.com/user-attachments/assets/c9ff9c09-dded-478e-88fc-c68044fe726e)
I selected the checkpoint from epoch 6 of the CoT phase for training Coconut, as described in the paper, which has an accuracy of about 40%. Then, in the latent CoT phase, I chose the checkpoint from epoch 6 of Coconut and was able to reproduce the results on GSM8K reported in the paper (paper: 34.1%, ours: 36.1%). However, I noticed that the loss curve increases after each phase, and the evaluation accuracy seems to decrease when using subsequent checkpoints. Is this a normal phenomenon?

**Comment by Ber666** [comment]
Hi, thanks for the question. Yes we observed the increasing loss too. I think that makes sense, since the task becomes inherently harder as the stage steps.

**Comment by PolinaDruzhinina** [comment]
Hi! I selected the checkpoint from the last epoch and the validation loss for the coconut training tends to increase. Did the validation loss increase during training in your case as well?

**Comment by Ber666** [comment]
> Hi! I selected the checkpoint from the last epoch and the validation loss for the coconut training tends to increase. Did the validation loss increase during training in your case as well?

Yes
